### PR TITLE
Implement ESLint configuration

### PR DESCRIPTION
**Pull request by xelav**
This PR are primarily for refactoring. It adds ESLint configuration to make sticking to some code guidelines much easier.
Important points of ESLint configuration:

- It derives from [Google's config](https://github.com/google/eslint-config-google)
- LF are preferred to CRLF
- 4 spaces for indentation instead of 2 spaces
- maximum line width is extended to 100 symbols

Other changes include:

- separating SongList component from MainPage
- making special reducer and action for search bar toggle

### CODE REVIEW

File: vanguarde/src/components/Footer.jsx [hunk 1]
```diff
@@ -14,42 +13,44 @@ export class Footer extends Component {
...
     }
 
-    handleFooterClick(e){
+    handleFooterClick(e) {
         this.footerClicks += 1;
-        if (this.footerClicks > 5){
-            let newFooterText = this.footerArray[(this.footerClicks - 5) % this.footerArray.length];
+        if (this.footerClicks > 5) {
+            const newFooterText = this.footerArray[(this.footerClicks - 5) %
+        this.footerArray.length];
```
Comment (kostmetallist) [inline vanguarde/src/components/Footer.jsx:30]: Are you sure it is valid indentation? This line locates at the same level as the `if`, what is weird for me.

File: vanguarde/src/components/SongCard.jsx [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
-                                <div className="col-md-6">
-                                    <div className="card-body">
-                                        <h4 className="card-title">{this.props.artist} -  {this.props.trackName} </h4>
-                                    </div>
+                <MDBCol md="10">
+                    <MDBCard className="align-middle mb-3 text-left">
+                        <div className="row no-gutters">
+                            <div className="col-md-1" style={{minWidth: '150px'}}>
+                                <img src={placeholder} alt="thumbnail"
+                                />
```
Comment (kostmetallist) [inline vanguarde/src/components/SongCard.jsx:23]: Shouldn't it be placed on the previous line?

File: vanguarde/src/components/SongCard.jsx [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
+                            </div>
+                            <div className="col-md-6">
+                                <div className="card-body">
+                                    <h4 className="card-title">
+                                        {this.props.artist} -  {this.props.trackName}
```
Comment (kostmetallist) [inline vanguarde/src/components/SongCard.jsx:28]: If this PR is targeted on the refactoring, then please remove this redundant spacing

File: vanguarde/src/serviceWorker.js [hunk 4]
```diff
@@ -11,131 +11,132 @@
...
-          console.log(
-            'This web app is being served cache-first by a service ' +
-              'worker. To learn more, visit https://bit.ly/CRA-PWA'
-          );
+                // Add some additional logging to localhost, pointing developers to the
+                // service worker/PWA documentation.
+                navigator.serviceWorker.ready.then(() => {
+                    console.log(
+                        'This web app is being served cache-first by a service ' +
+              'worker. To learn more, visit https://bit.ly/CRA-PWA',
```
Comment (kostmetallist) [inline vanguarde/src/serviceWorker.js:47]: Again, this indentation looks inappropriate for me

File: vanguarde/src/serviceWorker.js [hunk 5]
```diff
@@ -11,131 +11,132 @@
...
-              // Execute callback
-              if (config && config.onUpdate) {
-                config.onUpdate(registration);
-              }
-            } else {
-              // At this point, everything has been precached.
-              // It's the perfect time to display a
-              // "Content is cached for offline use." message.
-              console.log('Content is cached for offline use.');
+                            // Execute callback
```
Comment (kostmetallist) [inline vanguarde/src/serviceWorker.js:78]: It looks like I need help in understanding the principles of all that spacings :c

File: vanguarde/src/components/Footer.jsx (hunk 1 above)
Comment (xelav) [reply vanguarde/src/components/Footer.jsx]: Fixed this and all other issues that was brought up.

**Comment by xelav** [comment]
With nonexistent executive power, I temporarily overrule current guideline "Two approves minimum"

**Pull request merged**